ios/brave-ios/Sources/AIChat/ModelView/AIChatViewModel.swift

```diff
@@ -98,13 +98,19 @@ public class AIChatViewModel: NSObject, ObservableObject {
 
     // Initialize
     api = braveCore.aiChatAPI(with: self)
-    currentModel = api.currentModel
     models = api.models
     conversationHistory = api.conversationHistory
     suggestedQuestions = api.suggestedQuestions
     apiError = api.currentAPIError
     requestInProgress = api.isRequestInProgress
 
+    if let defaultModelKey = Preferences.AIChat.defaultAIModelKey.value {
+      currentModel = api.models.first(where: { $0.key == defaultModelKey }) ?? api.currentModel
+      changeModel(modelKey: defaultModelKey)
```

Comment:
`ConversationDriver` has its own mechanism for setting the default model from pref or varying via Premium status, which _could_ be async and race with this. At the moment it's not likely to be a bad race condition but that could change in the future. Could you at least set a TODO here to move to the shared pref for default model? We should sync it anyway...

Reply:
Fixed. I exported the preference here: https://github.com/brave/brave-core/pull/22627/commits/64dd821cc41128fc1ba0177800a31aa1df2fd737